guides/common/modules/proc_opening-required-ports.adoc

```diff
@@ -0,0 +1,83 @@
+:_mod-docs-content-type: PROCEDURE
+
+[id="opening-required-ports"]
+= Opening required ports
+
+For the components of {Project} architecture to communicate, ensure that the required network ports are open and free on the base operating system.
+You must also ensure that the required network ports are open on any network-based firewalls.
+
+[NOTE]
+====
+Some cloud solutions must be specifically configured to allow communications between machines because they isolate machines similarly to network-based firewalls.
+If you use an application-based firewall, ensure that the application-based firewall permits all applications that are listed in the tables and known to your firewall.
+If possible, disable the application checking and allow open port communication based on the protocol.
+====
+
+ifndef::satellite,orcharhino[]
+If you do not use `firewall-cmd` to configure the Linux firewall, implement using the command of your choice.
+endif::[]
+
+.Procedure
+. Optional: If you need to prevent the DHCP {SmartProxy} from pinging hosts to check for available IP addresses, disable DHCP IP address pinging:
++
+[options="nowrap", subs="+quotes,attributes"]
+----
+# {foreman-installer} --foreman-proxy-dhcp-ping-free-ip false
+----
++
+By default, a DHCP {SmartProxy} performs ICMP ping and TCP echo connection attempts to hosts in subnets with DHCP IPAM set to find out if an IP address considered for use is free.
+ifdef::katello,satellite,orcharhino[]
+ifeval::["{context}" == "{project-context}"]
+. Open the ports for clients on {ProjectServer}:
+endif::[]
+ifeval::["{context}" == "{smart-proxy-context}"]
+. Open the ports for clients on {SmartProxyServer}:
+endif::[]
++
+[options="nowrap"]
+----
+# firewall-cmd \
+--add-port="8000/tcp" \
+--add-port="9090/tcp"
+----
+endif::[]
+ifeval::["{context}" == "{project-context}"]
+. Allow access to services on {ProjectServer}:
+endif::[]
+ifeval::["{context}" == "{smart-proxy-context}"]
+. Allow access to services on {SmartProxyServer}:
+endif::[]
++
+[options="nowrap"]
+----
+# firewall-cmd \
+--add-service=dns \
+--add-service=dhcp \
+--add-service=tftp \
+--add-service=http \
+--add-service=https \
+ifndef::katello,satellite,orcharhino[]
+--add-service=foreman-proxy \
+endif::[]
+--add-service=puppetmaster
+----
```

Comment:
This command opens ports for the services that might not be required. Specifically DHCP and DNS were mentioned, which users will need to enable only when following https://docs.theforeman.org/3.15/Integrating_Provisioning_Infrastructure_Services/index-katello.html.

If I have more context (which services are required? what do we want to recommend and explain to users when installing a server/proxy?), I can update the step.

Reply:
We could go by the features that are enabled by default in the various builds?
Like Satellite has Provisioning enabled (does it?), so we do DNS/DHCP/TFTP here. But not Puppet, so no puppetmaster.
Foreman has Puppet by default, so it woud get puppetmaster.
Katello again no Puppet, and no idea about provisioning (I think the answer is "no")

Reply:
I introduced a different firewall-cmd command for each build @ekohl @evgeni Would this work? If so, I'm pretty sure the lists of services themselves are wrong; can you help me correct them?

Elsewhere @ekohl mentioned that at least for Katello, the list doesn't need to include DHCP, DNS, and TFTP because those are needed for https://docs.theforeman.org/3.15/Integrating_Provisioning_Infrastructure_Services/index-katello.html. However, that guide doesn't ask users to open the additional ports. Is that something that should be added there?